src/Orchard.Web/Modules/Orchard.Recipes/Services/RecipeExecutor.cs

```diff
@@ -56,54 +62,70 @@ public class RecipeExecutor : IRecipeExecutor
 
             try
             {
-                using (var stream = _fileSystem.GetFileInfo(recipeDescriptor.Location).CreateReadStream())
+                var fileInfo = _fileSystem.GetFileInfo(recipeDescriptor.Location);
+
+                var parsersForFileExtension = _recipeOptions
+                    .RecipeFileExtensions
+                    .Where(rfx => rfx.Key == Path.GetExtension(fileInfo.PhysicalPath));
+
+                using (var stream = fileInfo.CreateReadStream())
                 {
                     RecipeResult result = new RecipeResult { ExecutionId = executionId };
                     List<RecipeStepResult> stepResults = new List<RecipeStepResult>();
 
-                    await _recipeParser.ProcessRecipeAsync(stream, (recipe, recipeStep) =>
+                    foreach (var parserForFileExtension in parsersForFileExtension)
                     {
-                        // TODO, create Result prior to run
-                        stepResults.Add(new RecipeStepResult
+                        var recipeParser = _recipeParsers.First(x => x.GetType() == parserForFileExtension.Value);
```

Comment:
It's okay here, as we would expect that the parser would be registered. You wouldnt expect a parser for a file extension not to be registered in DI.